### ISSUE TITLE
google not finding the current sasview manual

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#1704]
A search on Google for ["sasview manual"](https://www.google.com/search?q=sasview+manual&oq=sasview+manual) doesn't return any link to the sasview manual on the first page, though some of them give it after a couple of links.

This is a google problem. Other search engines give the 5.0.3 manual as the first result (DDG, Bing, Yahoo).

Related issue: the 4.2.2 and 5.0.3 documentation are intermixed:
* 5.0.3 `http://www.sasview.org/docs/user/qtgui/Perspectives/Fitting/fitting.html`
* 4.2.2 `http://www.sasview.org/docs/user/sasgui/perspectives/fitting/fitting.html`

From the 4.2.2 docs if you go "up the hierarchy" to `http://www.sasview.org/docs/user/analysis.html` you end up in the 5.0.3 docs. This is presumably because files from the old version were not removed from the sasview/docs repo before the new version was added.

Docs for 3.0 are here: `http://www.sasview.org/sasview/user/models/model_functions.html`. I have no idea where that is hosted.

Suggestions:

* Reorganize docs repo so that versions are in `sasview.org/docs/5.0.3`, etc., rather than `docs/old_docs/5.0.3`
* Remove  `user`, `dev`, etc from top level of docs repo
* Link docs/current to the docs for the current release if possible
* Add "Docs" as a top level menu item on the sasview landing page pointing to the docs for the current release
* Pull the 3.x docs into the docs repo and remove the content from`www.sasview.org/sasview`
* Tell google to de-emphasize older doc versions

**Comment by smk78** [comment]
@pkienzle suggested
> * Reorganize docs repo so that versions are in `sasview.org/docs/5.0.3`, etc., rather than `docs/old_docs/5.0.3`

Done.

> * Pull the 3.x docs into the docs repo and remove the content from `www.sasview.org/sasview`

I've added the 3.1.2 docs to the docs repo, however these are different to those at `www.sasview.org/sasview` which claim to be 3.0.0. But if you install 3.0.0 you get a different set of docs again (3.0.0 was not released with Sphinx docs). Also, just exactly where `www.sasview.org/sasview` resides is proving a mystery!

**Comment by smk78** [comment]
@pkienzle commented:
> Docs for 3.0 are here: `http://www.sasview.org/sasview/user/models/model_functions.html`. I have no idea where that is hosted.

@wpotrzebowski says (copied from Slack #general):
It think it is coming from here https://github.com/SasView/sasview/blob/gh-pages/user/models/model_functions.html

**user/models/model_functions.html**
```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
```

I am not entirely sure but we may have github pages enabled in two places i.e for sasview repo and sasview.github.io. Let me take a look on this

Yes, that’s seem to be the case. We probably need to turn off sasview ones

I need to remove gh-pages from repository anyway: 
```
You can't disable GitHub Pages while having a gh-pages branch in this repository.
```

Ok, github-pages from sasview repo should be disabled now.

**Comment by smk78** [comment]
Docs that were at  https://github.com/SasView/sasview/blob/gh-pages are now in the docs repo as \old_docs\3.0.0

**Comment by smk78** [comment]
@pkienzle suggested:
> * Add "Docs" as a top level menu item on the sasview landing page pointing to the docs for the current release

Added "Documentation" under "Help". Points to documentation.md which has links to all versions of the documentation in the docs repo

**Comment by smk78** [comment]
Documentation now added as a top level menu item.
Documentation links withdrawn from Downloads page.
Downloads page overhauled.

**Comment by smk78** [comment]
@pkienzle commented:
>The 4.2.2 and 5.0.3 documentation are intermixed:
> 
> * 5.0.3 `http://www.sasview.org/docs/user/qtgui/Perspectives/Fitting/fitting.html`
> * 4.2.2 `http://www.sasview.org/docs/user/sasgui/perspectives/fitting/fitting.html`
> 
> From the 4.2.2 docs if you go "up the hierarchy" to `http://www.sasview.org/docs/user/analysis.html` you end up in the 5.0.3 docs. This is presumably because files from the old version were not removed from the sasview/docs repo before the new version was added.

This turned out to be because the 'SasView x.x.x documentation' tags in the Sphinx-built html doc files had got a bit munged for some reason. A few cases I recognised as my handiwork where I had done on-the-fly doc updates but not changed the program version (oops), but not the other almost 700 files!

All html files in the docs repo have now had their tag correctly attributed to the program version specified by the folder name, or as 5.0.3 for the top level.*

*For the interested, there is a neat feature in Notepad++ called 'Find in Files...' which will do a targetted replace.

**Comment by smk78** [comment]
Some Sphinx-built html doc files - not all - also contain in their header lines like this
```
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../../',
        VERSION:     '5.0.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
```
It is this VERSION string that determines the file paths when moving from one document to another. In the docs since version 4.2.2 around 180 files had legacy VERSION numbers from previous releases.

These have now also been updated.

**Comment by smk78** [comment]
@pkienzle commented:
> A search on Google for ["sasview manual"](https://www.google.com/search?q=sasview+manual&oq=sasview+manual) doesn't return any link to the sasview manual on the first page, though some of them give it after a couple of links.
> 
> This is a google problem. Other search engines give the 5.0.3 manual as the first result (DDG, Bing, Yahoo).
> 
> {We should} Tell google to de-emphasize older doc versions

I have added the words 'SasView Manual' to the FAQ. The answer points to both the 4.2.2 pdf and the new documentation page. In time, search engines might find this entry...

**Comment by smk78** [comment]
@pkienzle commented:
>The 4.2.2 and 5.0.3 documentation are intermixed:
> 
> * 5.0.3 `http://www.sasview.org/docs/user/qtgui/Perspectives/Fitting/fitting.html`
> * 4.2.2 `http://www.sasview.org/docs/user/sasgui/perspectives/fitting/fitting.html`

/docs/user/sasgui has been deleted from the top level of the docs repo (where the current qtgui docs reside). The contents are still available in /docs/old_docs/4.2.2/user/sasgui.

**Comment by smk78** [comment]
I think this issue has now been reasonably well dealt with so am going to close it.
